freqtrade/freqtradebot.py

```diff
@@ -395,9 +407,21 @@ def create_trade(self) -> bool:
 
         # running get_signal on historical data fetched
         # to find buy signals
+        if self.edge:
+            whitelist = self.edge.adjust(whitelist)
```

Comment:
I just thougt if this is the correct position for this...
assuming we run with 100 pairs - and edge removes 90 of them (leaving with 10 potential candidates)...

by placing edge pair selection here, it'll mean that we download 100x500 candles of ticker data in async - and then not use 90% of that.
i think this should be moved to update `_process` - before the `self.exchange.refresh_tickers` call... - but correct me if i'm wrong . 

Also - is it possible (or would it make sense) to only "partially" enable Edge - so it controls whitelist, but not stake amount?

Reply:
Need a tester for Edge with real money on binance ? I'm ready for it...

Reply:
@xmatthias you first point is valid. I am going to refactor it thanks ! 
but your second point is not. as its name implied, at the end it is about position sizing. actually in future what I want to do is to size the position against the size of your portfolio not against stake_amount. I am thinking that the fact that you ask this question means maybe I didn't explain position sizing well on the documentation ...

Also I have to admit that after refactoring tests I prefer merging this beast and then later adding new features if any.

@doftorul 
Thanks ! you can test it for sure. but please read the documentation first and tell me if it is clear enough. that would help us a lot.

Reply:
Applied the latest commits https://github.com/freqtrade/freqtrade/pull/1229/commits/4dcd15da1d7e393f593b78548f525511df6986c8
and there are no pairs left. I am testing with 100 pairs.

freqtrade.edge - INFO - Edge removed all pairs as no pair with minimum expectancy was found !
Nov 09 21:33:53 testrun.sh[27586]: 2018-11-09 21:33:53,367 - freqtrade.exchange - DEBUG - Refreshing klines for 0 pairs
Nov 09 21:33:53 testrun.sh[27586]: 2018-11-09 21:33:53,369 - freqtrade.exchange - DEBUG - Refreshing klines for 0 pairs
Nov 09 21:33:53 testrun.sh[27586]: 2018-11-09 21:33:53,371 - freqtrade.freqtradebot - WARNING - Unable to create trade: No currency pairs in whitelist